### PR TITLE
added repos and trackers to ProjectForm

### LINKED ISSUE #203: Add ability to add multiple source repositories on the project's form
When creating a project, users need to be able to one or more source repositories associated to their projects. On the create new projects form, add an Input field to capture an source repository url plus an ability to add more or remove already added ones.
## Expected Behavior

- Add an input field to add source repositories urls
- Validate to check that only valid urls are added via the input field
- Add a button to add more source repositories
- Add button to remove source repositories

## Current Behavior

The Create Project Page does not have ability to add source repositories
## Possible Solution

- Add a component for adding source repositories
- user regular expression to validate that valid url has been added to field before submit
- Add button to add more and remove source repositories
Below is the current implementation on website one that could be used as a guide
Screen Shot 2019-04-20 at 3 07 10 AM

## Your Environment

* Version used:
* Environment name and version (e.g. Chrome 39):
* Operating System and version (desktop or mobile):

### PR DESCRIPTION
**Pull request by jmpainter**
I found the cleanest implementation to be the use of redux-forms's FieldArray component.
The unit tests are passing, but I wanted to see what was thought about tests for the custom FieldArray component and the integration test for the ProjectForm

new file:   src/assets/CreateProjectPage.css - some styling needed for form additions
new file:   src/components/FieldGroup.js - custom redux-form FieldArray component
modified:   src/components/ProjectForm.js - added two redux-form FieldArray  components for repos and trackers
new file:   src/helpers/validators.js - validation for ProjectForm

fixes #203

### CODE REVIEW

File: src/tests/components/FieldGroup.test.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    expect(wrapper.find('Field').length).toEqual(2)
+  })
+
+  it('renders the first field with the correct label', () => {
+    const wrapper = shallow(
+      <FieldGroup fields={testFields} meta={meta} type={type} start={start} />
+    )
+    expect(wrapper.find('Field').filterWhere(field => {
+      return field.prop('label') === start + ' (primary)'
+    }).length).toEqual(1)
```
Comment (mattwr18) [inline src/tests/components/FieldGroup.test.js]: with Jest you can use `toHaveLength(1)` https://jestjs.io/docs/en/expect#tohavelengthnumber

File: src/tests/components/FieldGroup.test.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    }
+    const wrapper = shallow(
+      <FieldGroup fields={mockedFields} meta={meta} />
+    )
+    const addButton = wrapper.find('.field-group__add').first()
+    addButton.simulate('click')
+    expect(mockAdd).toHaveBeenCalled()
+  })
+
+  it('creates an add button with the correct type', () => {
```
Comment (mattwr18) [inline src/tests/components/FieldGroup.test.js:72]: should this test be moved before the one starting on line 64?

File: src/tests/components/FieldGroup.test.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  it('renders the second field with the correct label', () => {
+    const wrapper = shallow(
+      <FieldGroup fields={testFields} meta={meta} type={type} start={start} />
+    )
+    expect(wrapper.find('Field').filterWhere(field => {
+      return field.prop('label') === start + ' (2)'
+    }).length).toEqual(1)
+  })
+
+  it('adds a field', () => {
```
Comment (mattwr18) [inline src/tests/components/FieldGroup.test.js]: does it make sense to say `supports field addition` or something similar that portrays the idea that the user triggers the action?

File: src/tests/components/FieldGroup.test.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+  it('creates an add button with the correct type', () => {
+    const wrapper = mount(
+      <Provider store={store}>
+        <ProjectForm />
+      </Provider>
+    )
+    expect(wrapper.find('.ui.button.field-group__add').first().text()).toEqual('Add more repos')
+  })
+
+  it('removes a field', () => {
```
Comment (mattwr18) [inline src/tests/components/FieldGroup.test.js]: same as above :point_up:

File: src/tests/components/ProjectForm.test.js [hunk 5]
```diff
@@ -21,4 +23,50 @@ describe('ProjectForm', () => {
     form.simulate('submit')
     expect(props.handleSubmit).toHaveBeenCalledTimes(1)
   })
+
+  it('adds new repo fields', () => {
```
Comment (mattwr18) [inline src/tests/components/ProjectForm.test.js]: same as above :point_up:

File: src/tests/components/ProjectForm.test.js [hunk 6]
```diff
@@ -21,4 +23,50 @@ describe('ProjectForm', () => {
...
+    const wrapper = mount(
+      <Provider store={store}>
+        <ConnectedProjectForm />
+      </Provider>
+    )
+    wrapper.find('.ui.button.field-group__add').first().simulate('click')
+    expect(wrapper.find('Field[name="repos[0].value"]').length).toBeTruthy()
```
Comment (mattwr18) [inline src/tests/components/ProjectForm.test.js]: I am not a fan of `.toBeTruthy()` tests cause I feel like they are flaky... is there a way that you can test this without using it? 
here is something on the subject, if you want to have a look https://stackoverflow.com/questions/32615713/tobetrue-vs-tobetruthy-vs-tobetrue

I know there is `exists()` https://airbnb.io/enzyme/docs/api/ReactWrapper/exists.html... 
I'm assuming it doesn't have a predetermined length that we could use `toHaveLength(n)`?

File: src/tests/components/ProjectForm.test.js [hunk 7]
```diff
@@ -21,4 +23,50 @@ describe('ProjectForm', () => {
...
+    wrapper.find('.ui.button.field-group__add').first().simulate('click')
+    expect(wrapper.find('Field[name="repos[1].value"]').length).toBeTruthy()
```
Comment (mattwr18) [inline src/tests/components/ProjectForm.test.js]: can we avoid using multiple `expects` in the same it block?
https://softwareengineering.stackexchange.com/questions/7823/is-it-ok-to-have-multiple-asserts-in-a-single-unit-test

File: src/tests/components/ProjectForm.test.js [hunk 8]
```diff
@@ -21,4 +23,50 @@ describe('ProjectForm', () => {
...
+  })
+
+  it('removes repo fields', () => {
+    const wrapper = mount(
+      <Provider store={store}>
+        <ConnectedProjectForm />
+      </Provider>
+    )
+    wrapper.find('.ui.button.field-group__remove').first().simulate('click')
+    console.log(wrapper.debug())
```
Comment (mattwr18) [inline src/tests/components/ProjectForm.test.js]: just to point it out, cause everyone misses them sometimes :stuck_out_tongue:

File: src/tests/components/ProjectForm.test.js [hunk 9]
```diff
@@ -21,4 +23,50 @@ describe('ProjectForm', () => {
...
+  })
+
+  it('removes tracker fields', () => {
+    const wrapper = mount(
+      <Provider store={store}>
+        <ConnectedProjectForm />
+      </Provider>
+    )
+    wrapper.find('.ui.button.field-group__remove').at(2).simulate('click')
+    console.log(wrapper.debug())
```
Comment (mattwr18) [inline src/tests/components/ProjectForm.test.js]: another one... they're sneaky :rofl:

**Review by mattwr18 — CHANGES_REQUESTED** [review]
This is really great work @jmpainter!! 
I made some comments and suggestions, please feel free to let me know if there were motives I don't understand behind making the decisions you made, and/or push back on any changes you think are better not made. 

Also, it would be really helpful if you could get in the habit of uploading screenshots so that reviewers or just people interested in checking out what it looks like could do so without needing to pull down the branch and run the server/interact with the form.

Really impressed by how thorough you are on your `PR`s!!

**Comment by jmpainter** [comment]
Thanks for the suggestions. I've made these updates and added one more test to FieldGroup.

**Review by joaopapereira — APPROVED** [review]
LGTM

**Comment by mattwr18** [comment]
hey @jmpainter, I know I mentioned this is another `PR` review, but I'm not sure if you saw it. 
could you upload screen shots? it helps for us to give quicker feedback.

I think the code looks good, thank you for making the requested changes. :smile_cat: 
a couple things I noticed having pulled it down
Screenshot at 2019-05-07 15:15:28

Is there any way we can have the fields added between the `Delete` button and the `Add` button? like 
Screenshot at 2019-05-07 15:20:10

Also, I think we want to avoid just grey buttons, maybe we could make the `Remove` like https://react.semantic-ui.com/elements/button/#variations-negative
and the `Add` the `#34495E` blue color? what do you think? or if people think it looks better, maybe it could stay grey, but would look better with the `negative` color contrast?

**Comment by jmpainter** [comment]
I changed the placement of the add buttons and the colors of the add and remove buttons. I used the 'negative' class on the remove button. For the add buttons, the prop color="blue" was added. This looked like the way most native to React Semantic UI.
FieldGroup

**Review by mattwr18 — APPROVED** [review]
Great! looks good @jmpainter 
maybe we can open a separate ticket to move the `Create Project` button to the right so it's not directly below the others.
and is it just me or is the orange from our logo a little too similar to `negative` color from semantic ui react?
maybe we could consider changing it to the blue color from the navbar and footer?
separate ticket, though... as long as @joaopapereira is happy, let's get this merged!!

**Comment by jmpainter** [comment]
Ok, sounds good.

**Pull request merged**